lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -239,6 +239,17 @@ export class NormalComponent<
         if (existingPort) {
           existingPort.schematicSymbolPortDef = symPort
         } else {
+          if (
+            symPort.labels.includes("emitter") &&
+            opts.additionalAliases?.[`pin${pinNumber}`].includes("collector")
+          )
+            symPort.labels.splice(symPort.labels.indexOf("emitter"), 1)
+
+          if (
+            symPort.labels.includes("collector") &&
+            opts.additionalAliases?.[`pin${pinNumber}`].includes("emitter")
+          )
```

Comment:
this custom logic shouldn't be here in `NormalComponent` which is basically the base class for everything